File: KlirTechChallenge/angular/src/app/models/ShoppingCart.ts
import { Promotions } from './Product';

export class ShoppingCart {
  shoppingCartItems: ShoppingCartItem[];
  total: number;
}

export class ShoppingCartItem {
  item: string;
  quantity: number;
  price: number;
  total: number;
  promotion: Promotions;
  promotionApplied: string;
  constructor(item: string, quantity: number, price: number, promotion: Promotions, promotionApplied: string) {
    this.item = item;
    this.quantity = quantity;
    this.price = price;
    this.total = price;
    this.promotion = promotion;
    this.promotionApplied = promotionApplied;
  }
}

File: KlirTechChallenge/aspnet-core/Klir.TechChallenge.Web.Api/Data/Models/ShoppingCart.cs
using KlirTechChallenge.Web.Api.Data.Models;
using System.Collections.Generic;

namespace Klir.TechChallenge.Web.Api.Data.Models
{
    public class ShoppingCart
    {
        public List<ShoppingCartItem> ShoppingCartItems { get; set; }
        public double Total { get; set; }
    }

    public class ShoppingCartItem
    {
        public string Item { get; set; }
        public int Quantity { get; set; }
        public double Price { get; set; }
        public double Total { get; set; }
        public PromotionType Promotion { get; set; }
        public string PromotionApplied { get; set; }
    }
}
